### DISCUSSION TITLE
Styler for OJS chunks in QMD Files

### DISCUSSION THREAD

**Discussion by jack-davison** [posit-dev/positron#4196] [Feedback] [answered]
Excited to start using Positron!

Curious if there is, or if there are plans for, automatic styling of OJS chunks in QMD files?

e.g., something that would turn

````qmd
```{ojs}
Plot.plot({
marks: [
        Plot.dot(transpose(iris), {x: "Sepal.Length", y: "Petal.Width"})
        ]
})
```
````

into 

````qmd
```{ojs}
Plot.plot({
    marks: [
        Plot.dot(transpose(iris), {x: "Sepal.Length", y: "Petal.Width"})
    ]
})
```
````

Or similar. Can't get CTRL-SHIFT-A or CTRL-K CTRL-F to work!

Cheers

**Comment by juliasilge** [comment, marked as answer]
The solution to this will need to come from the Quarto extension, since it provides formatting for `.qmd` files. They have a command to format a cell:

![Screenshot 2024-07-31 at 12 42 07 PM](https://github.com/user-attachments/assets/2ee220c3-5b06-48b4-a002-accd12da9e9d)

Unfortunately though, that does not yet work for Observable chunks (it _does_ work for R and Python). Can you open an issue on the repo for the Quarto extension that you would like the "Quarto: Format Cell" command to work for Observable?
https://github.com/quarto-dev/quarto/issues

**Reply by jack-davison** [reply]
Thanks Julia, will do!
